app/celer-g4/RootIO.cc

```diff
@@ -62,76 +62,82 @@ HitRootIO::HitRootIO()
 
         file_.reset(TFile::Open(file_name_.c_str(), "recreate"));
         CELER_VALIDATE(file_->IsOpen(), << "failed to open " << file_name_);
-        tree_.reset(new TTree(
-            "Events", "Hit collections", this->SplitLevel(), file_.get()));
+        tree_.reset(new TTree(this->TreeName(),
+                              this->TreeName(),
+                              this->SplitLevel(),
+                              file_.get()));
     }
 }
 
-//---------------------------------------------------------------------------//
-//! Default destructor
-HitRootIO::~HitRootIO() = default;
-
 //---------------------------------------------------------------------------//
 /*!
- * Return the static thread local singleton instance>
+ * Return the static thread local singleton instance.
  */
-HitRootIO* HitRootIO::Instance()
+RootIO* RootIO::Instance()
 {
-    static G4ThreadLocal HitRootIO instance;
+    static G4ThreadLocal RootIO instance;
     return &instance;
 }
 
 //---------------------------------------------------------------------------//
 /*!
- * Write sensitive hits to output in the form of HitRootEvent.
+ * Write sensitive hits to output in the form of EventData.
+ * See celeritas/io/EventData.hh
  */
-void HitRootIO::WriteHits(G4Event const* event)
+void RootIO::Write(G4Event const* event)
 {
-    G4HCofThisEvent* hce = event->GetHCofThisEvent();
-    if (hce == nullptr)
+    auto const hit_cols = event->GetHCofThisEvent();
+    if (!hit_cols)
     {
         return;
     }
 
-    // Write the collection of sensitive hits into HitRootEvent
-    HitRootEvent hit_event;
-    hit_event.event_id = event->GetEventID();
-    for (int i = 0; i < hce->GetNumberOfCollections(); i++)
+    // Populate EventData using the collections of sensitive hits
+    EventData event_data;
+    event_data.steps.resize(detector_name_id_map_.size());
+
+    event_data.event_id = event->GetEventID();
+    for (int i = 0; i < hit_cols->GetNumberOfCollections(); i++)
```

Comment:
Use `range` here and for `j` below. Also consider renaming to `hc_id` and `hit_id`?